components/MarkdownEditor.tsx

```diff
@@ -0,0 +1,97 @@
+import "@uiw/react-md-editor/markdown-editor.css";
+import "@uiw/react-markdown-preview/markdown.css";
+
+import { Button, Icon } from "@equinor/eds-core-react";
+import MDEditor, { commands } from "@uiw/react-md-editor";
+import React, { useState } from "react";
+import { edit, check } from "@equinor/eds-icons";
+
+import { Typography } from "@equinor/eds-core-react";
+import rehypeSanitize from "rehype-sanitize";
+
+type PreviewType = "live" | "preview";
+
+export default function MarkdownEditor(props: {
+  canEdit: boolean;
+  defaultValue: string;
+  id: string;
+  label?: string;
+  onChange?: (value?: string) => void;
+}) {
+  const { canEdit, defaultValue, id, label = "", onChange } = props;
+  const [value, setValue] = useState(defaultValue);
+  const [preview, setPreview] = useState<PreviewType>("preview");
```

Comment:
What's the difference between canEdit, preview and disabled?

Consider simplifying as we talked about... Getting rid of previewType